### PR TITLE
Fix Error in newest GitExtension Version (v4) #28

### LINKED ISSUE #28: Plugin not working - Error in newest GitExtension Version (v4)
I installed the latest version of GitExtension yesterday (V4). Unfortunately an error is thrown as soon as I clicked the dropdown from SolutionRunner Plugin.

**Installed GitExtension Version:**
https://github.com/gitextensions/gitextensions/releases/tag/v4

**Error Scenario:**
SolutionRunner_GitExtensionV4_Error

**Error Detail:**
> - Git Extensions 4.0.0.15569
> - Build cba315df5366fca01240d93b6e5a5077fd5e6630
> - Git 2.38.1.windows.1
> - Microsoft Windows NT 10.0.19042.0
> - .NET 6.0.10
> - DPI 96dpi (no scaling)
> ```
> System.MissingMethodException: Method not found: '!!0 GitUIPluginInterfaces.ISettingsSource.GetValue(System.String, !!0, System.Func`2<System.String,!!0>)'.
>    at GitExtensions.SolutionRunner.PluginSettings.get_IsTopLevelSearchedOnly()
>    at GitExtensions.SolutionRunner.UI.SolutionListMenuItem.OnDropDownOpening(Object sender, EventArgs e) in D:\Development\Maraf\GitExtensions.SolutionRunner\src\GitExtensions.SolutionRunner\UI\SolutionListMenuItem.cs:line 40
>    at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
> ```

image
image
image

### PR DESCRIPTION
**Pull request by AngieKadnar**
Fixes https://github.com/maraf/GitExtensions.SolutionRunner/issues/28

### CODE REVIEW

**Comment by AngieKadnar** [comment]
Fixes #28

File: src/GitExtensions.SolutionRunner/UI/SolutionItemMenuItem.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using System.Windows;
```
Comment (maraf) [inline src/GitExtensions.SolutionRunner/UI/SolutionItemMenuItem.cs]: Is this one needed?

File: src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj [hunk 2]
```diff
@@ -1,9 +1,9 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
-
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
   <PropertyGroup>
-    <TargetFramework>net461</TargetFramework>
+    <TargetFramework>net7.0-windows</TargetFramework>
```
Comment (maraf) [inline src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj]: Git Extensions v4 targets net6.0

File: src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj [hunk 3]
```diff
@@ -1,9 +1,9 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
-
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
   <PropertyGroup>
-    <TargetFramework>net461</TargetFramework>
+    <TargetFramework>net7.0-windows</TargetFramework>
     <VersionPrefix>5.1.0</VersionPrefix>
     <PublishRepositoryUrl>true</PublishRepositoryUrl>
+	  <UseWindowsForms>true</UseWindowsForms>
```
Comment (maraf) [inline src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj]:
```suggestion
  <UseWindowsForms>true</UseWindowsForms>

```

File: src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj [hunk 4]
```diff
@@ -16,12 +16,14 @@
       <LastGenOutput>Resources.Designer.cs</LastGenOutput>
     </EmbeddedResource>
   </ItemGroup>
-
   <ItemGroup>
-    <PackageReference Include="GitExtensions.Extensibility" Version="0.1.*" />
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
+    <PackageReference Include="GitExtensions.Extensibility" Version="0.3.0.57" />
```
Comment (maraf) [inline src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj]:
```suggestion
    <PackageReference Include="GitExtensions.Extensibility" Version="0.3.*" />

```

File: src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj [hunk 5]
```diff
@@ -16,12 +16,14 @@
...
-
   <ItemGroup>
-    <PackageReference Include="GitExtensions.Extensibility" Version="0.1.*" />
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
+    <PackageReference Include="GitExtensions.Extensibility" Version="0.3.0.57" />
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers" Version="0.4.355802">
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="Microsoft.Windows.Compatibility" Version="7.0.0" />
```
Comment (maraf) [inline src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj]: Do we need these two?

File: src/GitExtensions.SolutionRunner/PluginSettings.cs [hunk 6]
```diff
@@ -44,27 +44,27 @@ internal class PluginSettings : IEnumerable<ISetting>
         /// <summary>
         /// Gets current value of <see cref="ExecutablePathProperty"/>.
         /// </summary>
-        public string ExecutablePath => source.GetValue(ExecutablePathProperty.Name, ExecutablePathProperty.DefaultValue, t => t);
+        public string ExecutablePath => source.GetValue(ExecutablePathProperty.Name);
```
Comment (maraf) [inline src/GitExtensions.SolutionRunner/PluginSettings.cs]: What happens if user doesn't provide a value? 
Does the default value still apply?

**Review by maraf — CHANGES_REQUESTED** [review]
Thanks for the contribution! Looks good with exception to some comments bellow.

File: src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj (hunk 5 above)
Comment (AngieKadnar) [reply src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj]: The first was a mistake which I forgot to remove. The Compatibility package reference is needed for some usings.

File: src/GitExtensions.SolutionRunner/PluginSettings.cs (hunk 6 above)
Comment (AngieKadnar) [reply src/GitExtensions.SolutionRunner/PluginSettings.cs]: I fixed it :)

File: src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj (hunk 2 above)
Comment (AngieKadnar) [reply src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj]: I downgrade it to net6

File: src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj [hunk 7]
```diff
@@ -1,9 +1,8 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
-
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
   <PropertyGroup>
-    <TargetFramework>net461</TargetFramework>
-    <VersionPrefix>5.1.0</VersionPrefix>
+    <TargetFramework>net6.0-windows</TargetFramework>
     <PublishRepositoryUrl>true</PublishRepositoryUrl>
+	<UseWindowsForms>true</UseWindowsForms>
```
Comment (maraf) [inline src/GitExtensions.SolutionRunner/GitExtensions.SolutionRunner.csproj]:
```suggestion
    <UseWindowsForms>true</UseWindowsForms>

```

**Review by maraf — APPROVED** [review]
Good job!

**Pull request merged**